core/src/main/javascript/msg/MessageBuilder.js

```diff
@@ -67,478 +67,53 @@
-                    e.setEntityAuthenticationData(entityAuthData);
-                    e.setUserIdToken(userIdToken);
-                    e.setUserAuthenticationData(userAuthData);
-                    e.setMessageId(requestMessageId);
-                }
-                throw e;
-            }
-        });
-    };
-
-    /**
-     * <p>Create a new message builder that will craft a new error message in
-     * response to another message. If the message ID of the request is not
-     * specified (i.e. unknown) then a random message ID will be generated.</p>
-     *
-     * @param {MslContext} ctx MSL context.
-     * @param {?number} requestMessageId message ID of request. May be null.
-     * @param {MslError} error the MSL error.
-     * @param {string} userMessage localized user-consumable error message. May be null.
-     * @param {{result: function(ErrorHeader), error: function(Error)}}
-     *        callback the callback that will receive the error header or any
-     *        thrown exceptions.
-     * @throws MslCryptoException if there is an error encrypting or signing
-     *         the message.
-     * @throws MslEntityAuthException if there is an error with the entity
-     *         authentication data.
-     * @throws MslMessageException if no entity authentication data was
-     *         returned by the MSL context.
-     */
-    var MessageBuilder$createErrorResponse = function MessageBuilder$createErrorResponse(ctx, requestMessageId, error, userMessage, callback) {
-        AsyncExecutor(callback, function() {
-            ctx.getEntityAuthenticationData(null, {
-                result: function(entityAuthData) {
-                    AsyncExecutor(callback, function() {
-                        // If we have the request message ID then the error response message ID
-                        // must be equal to the request message ID + 1.
-                        var messageId;
-                        if (requestMessageId != undefined && requestMessageId != null) {
-                            messageId = MessageBuilder$incrementMessageId(requestMessageId);
-                        }
-                        // Otherwise use a random message ID.
-                        else {
-                            messageId = MslUtils.getRandomLong(ctx);
-                        }
-                        var errorCode = error.responseCode;
-                        var internalCode = error.internalCode;
-                        var errorMsg = error.message;
-                        ErrorHeader.create(ctx, entityAuthData, messageId, errorCode, internalCode, errorMsg, userMessage, callback);
-                    });
-                },
-                error: callback.error,
-            });
-        });
-    };
+            }, self);
+        },
 
-    var MessageBuilder = module.exports = Class.create({
         /**
          * Create a new message builder with the provided tokens and key exchange
```

Comment:
Should probably change the text description of this method in both Java and JavaScript to say it is initializing the message builder instead of creating a new one.

Also looks like there is a typo in the `@param` list below that says `{MsaterToken}` and an extra space in `{MasterToken }`.

Reply:
Thanks Wes, I've addressed your comments around documentation bugs.